src/mysky/index.ts

```diff
@@ -0,0 +1,160 @@
+export type { CustomConnectorOptions } from "./connector";
+export { DacLibrary } from "./dac";
+
+import { PermCategory, PermHidden, Permission, PermType, PermWrite } from "skynet-interface-utils";
+
+import { Connector, CustomConnectorOptions } from "./connector";
+import { SkynetClient } from "../client";
+import { DacLibrary } from "./dac";
+import { RegistryEntry, setSignedEntry, SignedRegistryEntry } from "../registry";
+import { CustomGetJSONOptions, CustomSetJSONOptions, getOrCreateRegistryEntry, JsonData } from "../skydb";
+import { hexToUint8Array } from "../utils/string";
+import { Signature } from "../crypto";
+import { deriveDiscoverableTweak } from "./tweak";
+
+export async function loadMySky(
+  this: SkynetClient,
+  appPath: string,
+  customOptions?: CustomConnectorOptions
+): Promise<MySky> {
+  const mySky = await MySky.init(this, customOptions);
+  mySky.addPermissions(new Permission(appPath, PermHidden, PermWrite));
+
+  return mySky;
+}
+
+export const mySkyDomain = "skynet-mysky.hns";
+
+export class MySky {
+  // ============
+  // Constructors
+  // ============
+
+  constructor(protected connector: Connector, protected permissions: Permission[]) {}
+
+  static async init(client: SkynetClient, customOptions?: CustomConnectorOptions): Promise<MySky> {
+    const connector = await Connector.init(client, mySkyDomain, customOptions);
+
+    const domain = await client.extractDomain(window.location.hostname);
+    // TODO: Add requestor field to Permission?
+    // TODO: Are these permissions correct?
+    const perm = new Permission(domain, PermCategory.Hidden, PermType.Write);
+    const permissions = [perm];
+
+    return new MySky(connector, permissions);
+  }
+
+  // ==========
+  // Public API
+  // ==========
+
+  /**
+   * Loads the given DACs. Takes one or more instantiated DAC libraries.
+   */
+  async loadDacs(dac: DacLibrary, ...dacs: DacLibrary[]): Promise<void> {
+    if (dacs) {
+      dacs.unshift(dac);
+    } else {
+      dacs = [dac];
+    }
+
+    const promises: Promise<void>[] = [];
+    for (const dac of dacs) {
+      promises.push(this.loadDac(dac));
+    }
+
+    await Promise.all(promises);
+  }
+
+  async addPermissions(...permissions: Permission[]) {
+    this.permissions.push(...permissions);
+  }
+
+  async checkLogin(): Promise<boolean> {
+    return this.connector.connection.remoteHandle().call("checkLogin", this.permissions);
+  }
+
+  /**
+   * Destroys the mysky connection by:
+   *
+   * 1. Destroying the connected DACs,
+   *
+   * 2. Closing the bridge connection,
```

Comment:
maybe not mention the concept of a bridge now that it's gone?

Reply:
Yeah copy/paste error